### PR TITLE
Added: Log File

### LINKED ISSUE #149: Log file always available
### Abstract

Be able to always have the logs available with the information of the last run with `-v`

### Proposal

Always write the logs of the last run to `$XDG_CACHE_HOME`. Should also be able to be modifield with `--log-file=path` or `LOG_FILE=path`.

### Implementation

Just always initialize the log writter to the LOG_FILE and if the `-v` is used then use a `io.MultiWriter`.

### PR DESCRIPTION
**Pull request by xescugc**
Added a log file to always print the output to.

Closes #149

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,8 @@
   ([Issue #48](https://github.com/cycloidio/terracognita/issues/48))
 - Azure Virtual Desktop resources
   ([PR #145](https://github.com/cycloidio/terracognita/pull/145))
+- Log File to always write the last -v logs to
+  ([Issue #145](https://github.com/cycloidio/terracognita/issues/149))
```
Comment (tormath1) [inline CHANGELOG.md:16]:
```suggestion
  ([Issue #149](https://github.com/cycloidio/terracognita/issues/149))
```

File: provider/import.go [hunk 2]
```diff
@@ -93,7 +93,7 @@ func Import(ctx context.Context, p Provider, hcl, tfstate writer.Writer, f *filt
 				// we filter the error: if it's an error provider side, we continue
 				// the import but we print the error.
 				if errors.Is(err, errcode.ErrProviderAPI) {
-					fmt.Fprintf(out, "\nunable to import resource %s: %s\n", t, err.Error())
+					logger.Log("ms", fmt.Sprintf("unable to import resource %s: %s\n", t, err.Error()))
```
Comment (tormath1) [inline provider/import.go:96]: `"ms"` ?

File: cmd/root.go [hunk 3]
```diff
@@ -5,37 +5,53 @@ import (
...
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			err := os.MkdirAll(path.Dir(viper.GetString("log-file")), 0700)
+			if err != nil {
+				return err
+			}
+			logFile, err := os.OpenFile(viper.GetString("log-file"), os.O_APPEND|os.O_TRUNC|os.O_RDWR|os.O_CREATE, 0644)
+			if err != nil {
+				return err
+			}
+			closeOut = append(closeOut, logFile)
```
Comment (tormath1) [inline cmd/root.go:40]: no need to defer the call of `Close` in order to actually close the file handlers?

Comment (xescugc) [reply cmd/root.go:40]: We append all the `io.Closer` into this array so then when the CMD finishes can be closed all at once using `postRunEOutput` function.

**Comment by xescugc** [comment]
Done

File: cmd/root.go (hunk 3 above)
Comment (tormath1) [reply cmd/root.go:40]: Ok great, did not see the `postRunEOutput` method :)

**Comment by tormath1** [comment]
@xescugc except for the CHANGELOG, the rest LGTM

**Comment by xescugc** [comment]
Done

**Comment by tormath1** [comment]
Ok to RS :)

**Comment by xescugc** [comment]
Done

**Review by tormath1 — APPROVED** [review]

File: cmd/root.go [hunk 4]
```diff
@@ -115,6 +130,9 @@ func init() {
 	RootCmd.PersistentFlags().BoolP("debug", "d", false, "Activate the debug mode wich includes TF logs via TF_LOG=TRACE|DEBUG|INFO|WARN|ERROR configuration https://www.terraform.io/docs/internals/debugging.html")
 	_ = viper.BindPFlag("debug", RootCmd.PersistentFlags().Lookup("debug"))
 
+	RootCmd.PersistentFlags().String("log-file", path.Join(xdg.CacheHome, "terracognita", "log.log"), "Write the logs with -v to this destination")
```
Comment (xlr-8) [inline cmd/root.go:133]: I'd simply rename it to `terracognita.log` or `log(s)/terracognita.log` seems more standard.

**Review by xlr-8 — COMMENTED** [review]
Can be ignored, but I would change the default logfile name. Shouldn't more information be logged into the file, or is it meant to be in another PR?

File: cmd/root.go (hunk 4 above)
Comment (xescugc) [reply cmd/root.go:133]: I'll use `terracognita.log` as we do not have anything else there.

**Comment by xescugc** [comment]
Done

**Review by xlr-8 — APPROVED** [review]

**Comment by xescugc** [comment]
Done

**Pull request merged**